app/retail/templates/settings/privacy.html

```diff
@@ -17,6 +17,12 @@ <h5>{% trans "Privacy Preferences" %}</h5>
           <br>
           <small id="hide_profile_help" class="form__input-help">{% trans "If this option is chosen, we will remove your profile altogether" %}</small>
       </div>
+      <div>
+        <input type="checkbox" name="hide_wallet_address" id=hide_wallet_address {% if profile.hide_wallet_address %} checked="checked" {% endif %} {% if not is_logged_in %} disabled {% endif %} >
+        <label class="form__label" for="hide_wallet_address">{% trans "Hide my wallet address" %}</label>
+        <br>
+        <small id="hide_wallet_address_help" class="form__input-help">{% trans "If this option is chosen, we will remove your wallet address" %}</small>
```

Comment:
@owocki should we add some disclaimer type language to add "except those you've previously done business with"?

Reply:
agree with Dan we should put change that to '"will be hidden or will not be publicly visible" something like that, since we are not removing it for real.